emcc.py

```diff
@@ -2064,7 +2026,9 @@ def compile_source_file(i, input_file):
         cmd = get_clang_command_asm(input_file)
       else:
         cmd = get_clang_command(input_file)
-      cmd += ['-c', '-o', output_file]
+      if not compile_only:
+        cmd += ['-c']
```

Comment:
This seems counterintuitive. I would have expected `-c` to be used when we only want to compile. Why is the condition `not compile_only`? Also, where does `compile_only` come from?

Reply:
Agree its confusing.   `compile_only` basically means that that user specified `-c` themselves so we don't need to add it:

```compile_only = has_dash_c or has_dash_S or has_dash_E```

I changed to `if not has_dash_c:` which is more correct and more readable.